in_gdummy.go

```diff
@@ -17,18 +21,85 @@ type gdummyPlugin struct {
 	counterSuccess metric.Counter
 	counterFailure metric.Counter
 	log            plugin.Logger
+	gen            dataGen
+	dummyMsg       map[string]string
+	template       *LogGenerator
+	rate           time.Duration
 }
 
+type dataGen uint
+
+const (
+	genDummy dataGen = iota
+	genTemplate
+)
+
 func (plug *gdummyPlugin) Init(ctx context.Context, fbit *plugin.Fluentbit) error {
 	plug.counterSuccess = fbit.Metrics.NewCounter("operation_succeeded_total", "Total number of succeeded operations", "gdummy")
 	plug.counterFailure = fbit.Metrics.NewCounter("operation_failed_total", "Total number of failed operations", "gdummy")
 	plug.log = fbit.Logger
 
+	if msg := fbit.Conf.String("dummy"); msg != "" {
+		if err := json.Unmarshal([]byte(msg), &plug.dummyMsg); err != nil {
+			return err
+		}
+	}
+
+	switch fbit.Conf.String("datagen") {
+	case "":
+		fallthrough
+	case "dummy":
+		plug.gen = genDummy
+		if plug.dummyMsg == nil {
+			return errors.New("invalid config: 'dummy' not set with datagen=dummy")
+		}
+	case "template":
+		plug.gen = genTemplate
+		fieldsStr := fbit.Conf.String("fields")
+		if fieldsStr == "" {
+			return errors.New("invalid config: 'fields' not set with datagen=template")
+		}
+		fieldsSeparated := strings.Split(fieldsStr, ",")
+		fields := make([]string, 0, len(fieldsSeparated))
+		for _, field := range fieldsSeparated {
+			field = strings.TrimSpace(field)
+			if field != "" {
+				fields = append(fields, field)
+			}
+		}
+		if len(fields) == 0 {
+			return errors.New("invalid config: 'fields' list is empty with datagen=template")
+		}
+
+		plug.template = NewLogGenerator()
+		for _, field := range fields {
+			fieldValue := fbit.Conf.String(field)
```

Comment:
I'm worried about this "dynamic" config in where the config schema is unknown.
I think I would just prefer if the template evaluates to a JSON string (At least for now since we don't support reading objects from the YAML config).

```yaml
pipeline:
  inputs:
    - name: datagen
      template: |-
        {
          "id": "{{UUID}}",
          "host": "{{IPv4Address}}",
          "userAgent": "{{UserAgent}}",
          "version": "{{AppVersion}}",
          "date": "{{PastDate}}"
        }
```

Reply:
I agree with this, for sure